docs/coordinates/inplace.rst

```diff
@@ -0,0 +1,53 @@
+In Place Modification of Coordinates
+====================================
+
+Coordinates are generally considered to be immutable. If you want to create
+another coordinate frame with different data you should use
+`~astropy.coordinates.BaseCoordinateFrame.realize_frame`, this is the safest way
+to change the data in a frame as it creates a new frame with that data.
+
+Due to the fact that this creates a new frame it is relatively slow, and in some
+situations it is required that the data is replaced inside a frame with minimal
+changes to the frame for maximum speed. This modification can be done by
+modifiying the values of the representation data as follows::
+
+    >>> import astropy.units as u
+    >>> from astropy.coordinates import SkyCoord
+    >>> c = SkyCoord([1,2],[3,4], unit='deg')
+    >>> c.data.lon[()] = [10, 20] * u.deg
+    >>> c
+    <SkyCoord (ICRS): (ra, dec) in deg
+        [( 10.,  3.), ( 20.,  4.)]>
+
+This changes the longitude values for the frame, however
+`~astropy.coordinates.SkyCoord` and `~astropy.coordinates.BaseCoordinateFrame`
+cache representations of the data held within the frame as they are calculated
+to speed up future operations of different representations of the same data. If
```

Comment:
Also had trouble with this, both understanding what it says, *and* the fact that it specifically mentions `['representation']` but not anything else that might be cached.  So I'd change it to:

  ```
This changes the longitude values of the frame. Unfortunately, doing just this introduces 
problems: `~astropy.coordinates.SkyCoord` and `~astropy.coordinates.BaseCoordinateFrame` 
cache various kinds of information to speed up some repeated operations.  So we need to tell 
the cache that it should be cleared so that it can be re-calculated from the new data.  
This can be achieved by doing::

    >>> del c.cache.clear()

or::

    >>> del c.frame.cache